src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorToolEnhanced.java

```diff
@@ -0,0 +1,1402 @@
+import org.opensearch.ml.common.utils.ToolUtils;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.search.aggregations.AggregationBuilder;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.timeseries.AnalysisType;
+import org.opensearch.timeseries.model.Feature;
+import org.opensearch.timeseries.model.IntervalTimeConfiguration;
+import org.opensearch.timeseries.model.TimeConfiguration;
+import org.opensearch.timeseries.transport.JobRequest;
+import org.opensearch.timeseries.transport.SuggestConfigParamRequest;
+import org.opensearch.timeseries.transport.SuggestConfigParamResponse;
+import org.opensearch.timeseries.transport.ValidateConfigRequest;
+import org.opensearch.timeseries.transport.ValidateConfigResponse;
+import org.opensearch.transport.client.Client;
+
+import com.google.common.collect.ImmutableMap;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Enhanced tool for creating anomaly detectors with LLM-assisted configuration and validation.
+ * Analyzes index mappings, generates detector configurations using LLM, validates through multiple phases,
+ * and automatically creates and starts detectors for multiple indices.
+ * 
+ * Usage:
+ * 1. Register agent:
+ * POST /_plugins/_ml/agents/_register
+ * {
+ *   "name": "AnomalyDetectorEnhanced",
+ *   "type": "flow",
+ *   "tools": [
+ *     {
+ *       "name": "create_anomaly_detector_enhanced",
+ *       "type": "CreateAnomalyDetectorToolEnhanced",
+ *       "parameters": {
+ *         "model_id": "model-id",
+ *         "model_type": "CLAUDE"
+ *       }
+ *     }
+ *   ]
+ * }
+ * 2. Execute agent:
+ * POST /_plugins/_ml/agents/{agent_id}/_execute
+ * {
+ *   "parameters": {
+ *     "input": ["ecommerce-data", "server-logs"]
+ *   }
+ * }
+ * 3. Result: detector creation status for each index
+ * {
+ *   "ecommerce-data": {
+ *     "status": "success",
+ *     "detectorId": "abc123",
+ *     "detectorName": "ecommerce-data-detector-xyz",
+ *     "createResponse": "Detector created successfully",
+ *     "startResponse": "Detector started successfully"
+ *   },
+ *   "server-logs": {
+ *     "status": "failed_validation",
+ *     "error": "Insufficient data for model training"
+ *   }
+ * }
+ */
+@Log4j2
+@Setter
+@Getter
+@ToolAnnotation(CreateAnomalyDetectorToolEnhanced.TYPE)
+public class CreateAnomalyDetectorToolEnhanced implements WithModelTool {
+    public static final String TYPE = "CreateAnomalyDetectorToolEnhanced";
+
+    private static final String DEFAULT_DESCRIPTION =
+        "Enhanced tool for creating anomaly detector configurations. Takes an index name, extracts the index mappings, and uses LLM to generate complete detector JSON configurations ready for the create detector API.";
+
+    private static final String EXTRACT_INFORMATION_REGEX =
+        "(?s).*\\{category_field=([^|]*)\\|aggregation_field=([^|]*)\\|aggregation_method=([^|]*)\\|interval=([^}]*)}.*";
```

Comment:
This seems to be a really strict regex that would fail on things like whitespace or the keys being in a different order (and cause retries which lose the benefit of the optimal date field.).  

Given we're just extracting characters between known strings, it seems there's likely a better way of iterating between the `|` characters, splitting on `=`, and trimming the key/value into a map?